protocols/mdns/src/lib.rs

```diff
@@ -49,8 +49,12 @@ pub use crate::behaviour::TokioMdns;
 
 /// The DNS service name for all libp2p peers used to query for addresses.
 const SERVICE_NAME: &[u8] = b"_p2p._udp.local";
+/// `SERVICE_NAME` as a Fully Qualified Domain Name.
+const SERVICE_NAME_FQDN: &str = "_p2p._udp.local.";
```

Comment:
`trust-dns` parses `Messages` as [`FQDN`](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) therefore the trailing dot.

Reply:
This might be an issue in the spec: https://github.com/libp2p/specs/blob/master/discovery/mdns.md

Reply:
Thanks for referring this Thomas, the PR is only using `trust-dns` to parse incoming messages, dns messages sent are still constructed with [`build_query`](https://github.com/libp2p/rust-libp2p/blob/313ed588fec84c4307998f04282b708af70d9012/protocols/mdns/src/behaviour/iface/dns.rs#L72) [`build_query_response`](https://github.com/libp2p/rust-libp2p/blob/313ed588fec84c4307998f04282b708af70d9012/protocols/mdns/src/behaviour/iface/dns.rs#L106), and [`build_service_discovery_response`](https://github.com/libp2p/rust-libp2p/blob/313ed588fec84c4307998f04282b708af70d9012/protocols/mdns/src/behaviour/iface/dns.rs#L168) which use the non `FQDN` names, so we should be ok right?

Reply:
But if `trust-dns` requires a dot at the end, doesn't this mean we are now not spec-compliant and can't parse the DNS messages from other implementations?

Reply:
`trust-dns` [follows the spec](http://www.dns-sd.org/trailingdotsindomainnames.html), it doesn't require a dot at the end, it puts it on constructed [`Message`](https://docs.rs/trust-dns-proto/0.22.0/trust_dns_proto/op/message/struct.Message.html)'s cause they are FQDN's. 
If you try constructing a `Message` from a name with a trailing dot it will not be valid, see [here](https://github.com/jxs/trust-dns-fqdns/blob/main/src/main.rs#L72). We could also [`Name::set_fqdn`](https://docs.rs/trust-dns-proto/0.22.0/trust_dns_proto/rr/domain/struct.Name.html#method.set_fqdn) as `false`, and it will lose the trailing dot.